libs/features/mas/src/mas-commerce-service.js

```diff
@@ -167,15 +178,56 @@ export class MasCommerceService extends HTMLElement {
     refreshOffers() {
         this.flushWcsCache();
         document
-            .querySelectorAll('span[is="inline-price"],a[is="checkout-link"]')
+            .querySelectorAll(SELECTOR_MAS_ELEMENT)
             .forEach((el) => el.requestUpdate(true));
         this.log.debug('Refreshed WCS offers');
+        this.logFailedRequests();
     }
 
     refreshFragments() {
         this.flushWcsCache();
         document.querySelectorAll('aem-fragment').forEach((el) => el.refresh());
         this.log.debug('Refreshed AEM fragments');
+        this.logFailedRequests();
+    }
+
+    /**
+     * Logs failed network requests related to AEM fragments and WCS commerce artifacts.
+     * Identifies failed resources by checking for zero transfer size, zero duration,
+     * response status less than 200, or response status greater than or equal to 400.
+     * Only logs errors if any of the failed resources are fragment or commerce artifact requests.
+     */
+    /* c8 ignore next 21 */
```

Comment:
we could probably play around with nock here, but yeah, another PR :)

Reply:
I think nock is for nodejs only, I have checked it after having seen you using in the endpoint tests.